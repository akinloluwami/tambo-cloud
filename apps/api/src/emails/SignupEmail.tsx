import React from "react";
import { Body, Container, Hr, Html, Link, Preview, Text } from "@react-email/components";

export interface SignupEmailProps {
  firstName?: string | null;
  docsUrl: string;
  unsubscribeUrl: string;
}

/*
 This component renders both HTML and plain-text automatically via @react-email/render.
 The test suite renders the component to verify the markup doesn’t unexpectedly change.
*/
export default function SignupEmail({
  firstName,
  docsUrl,
  unsubscribeUrl,
}: SignupEmailProps) {
  const greetingName = firstName?.trim() || "there";

  return (
    <Html>
      <Preview>Welcome to tambo – let’s get building</Preview>
      <Body
        style={{
          fontFamily:
            'ui-sans-serif, system-ui, -apple-system, "Segoe UI", sans-serif',
          margin: 0,
          padding: 0,
          backgroundColor: "#ffffff",
        }}
      >
        <Container style={{ maxWidth: "600px", margin: "0 auto", padding: "24px" }}>
          <Text style={{ fontSize: 18, marginBottom: 16 }}>welcome, {greetingName}!</Text>

          <Text style={{ fontSize: 16, margin: "16px 0" }}>
            Thanks for signing up!
          </Text>

          <Text style={{ fontSize: 16 }}>1) run: <code>npx create-tambo@latest</code></Text>
          <Text style={{ fontSize: 16 }}>
            2) quick-start guide: <Link href={docsUrl}>{docsUrl}</Link>
          </Text>

          <Text style={{ fontSize: 16, margin: "24px 0" }}>
            questions? reply to this email.
          </Text>

          <Text style={{ fontSize: 16, marginTop: 24 }}>— michael & the tambo team</Text>

          <Hr style={{ margin: "32px 0", borderColor: "#e5e7eb" }} />

          <Text style={{ fontSize: 12, color: "#6b7280" }}>
            Don’t want these emails?
            {" "}
            <Link href={unsubscribeUrl}>Unsubscribe</Link>
          </Text>
        </Container>
      </Body>
    </Html>
  );
}

// Helper to generate the plain-text version, exported for convenience.
export function renderSignupEmailText({ firstName, docsUrl, unsubscribeUrl }: SignupEmailProps) {
  const greetingName = firstName?.trim() || "there";
  return `welcome, ${greetingName}!\n\nThanks for signing up!\n\n1) run: npx create-tambo@latest\n2) quick-start guide: ${docsUrl}\n\nquestions? reply to this email.\n\n— michael & the tambo team\n\nUnsubscribe: ${unsubscribeUrl}\n`;
}
