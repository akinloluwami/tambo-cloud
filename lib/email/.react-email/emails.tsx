import { WelcomeEmail } from '../templates/welcome-email';
import { NotificationEmail } from '../templates/notification-email';

export const emails = [
  {
    name: 'Welcome Email',
    component: (
      <WelcomeEmail
        firstName="John Doe"
        projectName="My Amazing Project"
        loginUrl="https://tambo.co/login"
      />
    ),
  },
  {
    name: 'Welcome Email (Minimal)',
    component: (
      <WelcomeEmail
        firstName="Jane"
      />
    ),
  },
  {
    name: 'Notification Email',
    component: (
      <NotificationEmail
        title="New Feature Available"
        message="We're excited to announce a new feature that will help you build better applications. Check it out and let us know what you think!"
        actionUrl="https://tambo.co/features"
        actionText="View Feature"
        userName="John Doe"
      />
    ),
  },
  {
    name: 'Notification Email (Simple)',
    component: (
      <NotificationEmail
        title="Reminder"
        message="This is a simple notification message without any action buttons."
        userName="Jane Smith"
      />
    ),
  },
];