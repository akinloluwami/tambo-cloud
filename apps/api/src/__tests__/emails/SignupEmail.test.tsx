import { renderSignupEmailText } from "../../emails/SignupEmail";

describe("SignupEmail template", () => {
  it("matches snapshot", () => {
    const text = renderSignupEmailText({
      firstName: "Taylor",
      docsUrl: "https://docs.tambo.co/quick-start",
      unsubscribeUrl: "https://tambo.co/unsub?id=123",
    });

    expect(text).toMatchSnapshot();
  });
});
