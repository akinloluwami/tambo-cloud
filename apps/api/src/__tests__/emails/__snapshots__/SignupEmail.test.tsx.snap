// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`SignupEmail template matches snapshot 1`] = `Promise {}`;
